File: Java/2024_cursoemvideo_poo/aula04_get_set_constr/src/Caneta.java
public class Caneta {
    private String modelo;
    private float ponta;
    private boolean tampada;

    //public Caneta(String m, float p, boolean t) {
    //    this.modelo = m;
    //    this.ponta = p;
    //    this.tampada = t;
    //}

    public Caneta(String modelo, float ponta, boolean tampada) {
        this.modelo = modelo;
        this.ponta = ponta;
        this.tampada = tampada;
    }

    //MÉTODOS ESPECIAIS
    public String getModelo() {
        return this.modelo;
    }
    public void setModelo(String m) {
        this.modelo = m;
    }
    public float getPonta() {
        return this.ponta;
    }
    public void setPonta(float p) {
        this.ponta = p;
    }

    public boolean isTampada() {
        return tampada;
    }

    public void setTampada(boolean tampada) {
        this.tampada = tampada;
    }

    public void status() {
        System.out.println("O modelo é "+modelo);
        System.out.println("A ponta é "+ponta);
        System.out.println("A caneta está tampada:"+tampada);
    }
}

File: PHP/2016_cursoemvideo_poo/aula04_get_set_construct/Caneta.php
<?php 
class Caneta1 
// troquei o nome da classe pq documentos anteriores tavam buscando a classe Caneta aqui.
{
    private $modelo;
    private $ponta;
    private $tampada;
    
    public function getModelo() {
        return $this -> modelo;
    }

    public function setModelo($m) {
        $this->modelo = $m;
    }

    public function getPonta() {
        return $this -> ponta;
    }

    public function setPonta($p) {
        $this->ponta = $p;
    }

    /**
     * Get the value of tampada
     */
    public function getTampada()
    {
        return $this -> tampada;
    }

    /**
     * Set the value of tampada
     *
     * @return  self
     */ 
    public function setTampada($tampada)
    {
        $this -> tampada = $tampada;

        return $this;
    }
}

class Caneta2 {
    private $modelo;
    private $ponta;
    private $tampada;
    
    //Contruct
    public function __construct($m,$p,$t)
    {
     $this -> modelo = $m;
     $this -> ponta = $p;   
     $this -> tampada = $t;
    }

    public function getModelo() {
        return $this -> modelo;
    }
    public function getPonta() {
        return $this -> ponta;
    }
    public function getTampada() {
        return $this -> tampada;
    }
}
?>